src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,309 @@
+
+// Number of attempts to fetch the data before displaying an error.
+const FETCH_RETRIES = 3;
+
+/**
+ * Specifies the form of component props.
+ */
+interface PostDetailsSectionProps {
+	data: PostPerformanceData;
+}
+
+/**
+ * Outputs the current post's details or shows an error message on failure.
+ */
+function CurrentPostDetails() {
+	const [ loading, setLoading ] = useState<boolean>( true );
+	const [ error, setError ] = useState( null );
+	const [ postDetailsData, setPostDetails ] = useState<PostPerformanceData>( null );
+	const provider = new CurrentPostDetailsProvider();
+
+	useEffect( () => {
+		const fetchPosts = async ( retries: number ) => {
+			provider.getCurrentPostDetails()
+				.then( ( result ) => {
+					setPostDetails( result );
+					setLoading( false );
+				} )
+				.catch( async ( err ) => {
+					if ( retries > 0 ) {
+						await new Promise( ( r ) => setTimeout( r, 500 ) );
+						await fetchPosts( retries - 1 );
+					} else {
+						setError( err );
+						setLoading( false );
+					}
+				} );
+		};
+
+		setLoading( true );
+		fetchPosts( FETCH_RETRIES );
+	}, [] );
+
+	if ( error ) {
+		// "Soft" error for which we do not want to show an "Error:" prefix.
+		if ( 'string' === typeof error ) {
+			return <p>{ error }</p>;
+		}
+
+		// Error coming from apiFetch.
+		if ( error?.message ) {
+			return <p>{ __( 'Error:', 'wp-parsely' ) } { error.message }</p>;
+		}
+
+		// Error coming from the WordPress REST API.
+		const errorMessage = JSON.stringify( error ).match( /\[\"(.*?)\"\]/ )[ 1 ];
+		return <p>{ __( 'Error:', 'wp-parsely' ) } { errorMessage }</p>;
+	}
+
+	return (
+		<>
+			{ loading ? <Spinner /> : <CurrentPostDetailsSections data={ postDetailsData } /> }
+		</>
+	);
+}
+
+/**
+ * Outputs all the "Current Post Details" sections.
+ *
+ * @param {PostDetailsSectionProps} props Props needed to populate the sections.
+ */
+function CurrentPostDetailsSections( props: PostDetailsSectionProps ) {
+	const data = props.data;
+
+	return (
+		<div className="current-post-details-panel">
+			{ DataPeriodSection( data ) }
+			{ GeneralPerformanceSection( data ) }
+			{ ReferrerTypesSection( data ) }
+			{ TopReferrersSection( data ) }
+			{ ActionsSection( data ) }
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Period" section, which denotes the period for which data is
+ * shown.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function DataPeriodSection( data: PostPerformanceData ) {
+	const period = data.period;
+
+	// Get the date (in short format) on which the period starts.
+	const periodStart = Intl.DateTimeFormat(
+		document.documentElement.lang,
+		{ month: 'short', day: 'numeric' }
+	).format( new Date( period.start ) );
+
+	return (
+		<div className="section period">
+			{ __( 'Last ', 'wp-parsely' ) + period.days + __( ' Days ', 'wp-parsely' ) }
+			<span>
+				{ __( '(', 'wp-parsely' ) + periodStart + __( ' - Today)', 'wp-parsely' ) }
```

Comment:
Is it possible to make these strings combined, with an interpolated / placeholder? It will make the translations easier on translators.

Reply:
I have improved our i18n in [this commit](https://github.com/Parsely/wp-parsely/pull/1093/commits/cadb922761d967adb5f5af6196fd2bd4e548d508). Marking this and your other related remarks as resolved, but feel free to comment if something else should be needed.